ocaml/xenopsd/c_stubs/xenctrlext_stubs.c

```diff
@@ -394,6 +409,127 @@ CAMLprim value stub_xenctrlext_cputopoinfo(value xch)
 	CAMLreturn(result);
 }
 
+CAMLprim value stub_xenforeignmemory_open(value logger, value open_flags)
+{
+        CAMLparam2(logger, open_flags);
+        struct xentoollog_logger *log_handle = NULL;
+        struct xenforeignmemory_handle *fmem;
+        CAMLlocal1(result);
+
+        if(Is_some(logger)) {
+                log_handle = Xtl_val(Some_val(logger));
+        }
+
+        fmem = xenforeignmemory_open(log_handle, Int_val(open_flags));
+
+        if(fmem == NULL) {
+                caml_failwith("Error when opening foreign memory handle");
+        }
+
+        result = caml_alloc(1, Abstract_tag);
```

Comment:
Not sure which way around the order would be better here, but if this raises an exception (e.g. OutOfMemory) then we leak a xenforeignmemory handle.
If we allocate first then at worst we have allocated an abstract tag that we do not use, but the GC should collect it, so perhaps doing the caml_alloc first would be better.

Reply:
This is C code, so how can there be an exception? The OCaml code has a `defer` that should ensure the handle is closed.

Reply:
The surprising thing is that for all the warnings there are in the [manua](https://ocaml.org/manual/intfc.html) this is not among them. Maybe this is exceedingly rare in the real world, but it still feels suspect to have this behaviour.